tests/models/test_queryparams.py

```diff
@@ -18,36 +28,41 @@ def test_queryparams():
     assert dict(q) == {"a": "456", "b": "789"}
     assert str(q) == "a=123&a=456&b=789"
     assert repr(q) == "QueryParams('a=123&a=456&b=789')"
-    assert QueryParams({"a": "123", "b": "456"}) == QueryParams(
-        [("a", "123"), ("b", "456")]
-    )
-    assert QueryParams({"a": "123", "b": "456"}) == QueryParams("a=123&b=456")
-    assert QueryParams({"a": "123", "b": "456"}) == QueryParams(
-        {"b": "456", "a": "123"}
-    )
-    assert QueryParams() == QueryParams({})
-    assert QueryParams([("a", "123"), ("a", "456")]) == QueryParams("a=123&a=456")
-    assert QueryParams({"a": "123", "b": "456"}) != "invalid"
-
-    q = QueryParams([("a", "123"), ("a", "456")])
-    assert QueryParams(q) == q
-
-
-def test_queryparam_types():
-    q = QueryParams({"a": True})
-    assert str(q) == "a=true"
-
-    q = QueryParams({"a": False})
-    assert str(q) == "a=false"
 
-    q = QueryParams({"a": ""})
-    assert str(q) == "a="
 
-    q = QueryParams({"a": None})
-    assert str(q) == "a="
+@pytest.mark.parametrize(
+    "one, other",
+    [
+        [
+            QueryParams({"a": "123", "b": "456"}),
+            QueryParams([("a", "123"), ("b", "456")]),
+        ],
+        [QueryParams({"a": "123", "b": "456"}), QueryParams("a=123&b=456")],
+        [QueryParams({"a": "123", "b": "456"}), QueryParams({"b": "456", "a": "123"})],
+        [QueryParams(), QueryParams({})],
+        [QueryParams([("a", "123"), ("a", "456")]), QueryParams("a=123&a=456")],
+        [
+            QueryParams([("a", "123"), ("a", "456")]),
+            QueryParams(QueryParams([("a", "123"), ("a", "456")])),
+        ],
+    ],
+)
+def test_queryparams_equality(one, other):
+    assert one == other
 
-    q = QueryParams({"a": 1.23})
-    assert str(q) == "a=1.23"
 
-    q = QueryParams({"a": 123})
-    assert str(q) == "a=123"
+@pytest.mark.parametrize(
+    "source, output",
+    [
+        ({"a": True}, "a=true"),
+        ({"a": False}, "a=false"),
+        ({"a": ""}, "a="),
+        ({"a": None}, "a="),
```

Comment:
I still don't agree with this behavior. There's no way to represent just `a` in a query string without `=` and it's one of the [most commented issues for Requests](https://github.com/psf/requests/issues/2651) there's definitely a need.

Reply:
I see the need in there, yes. My safest bet is that we could use `None` for this.

But Requests also uses it to allow "unsetting" query params from the client for a specific request. Do we want to support that use case too? Manually omitting the key by rebuilding the `params` dict without the target value wouldn't work since we merge query params with the client's.

(I rarely wish we had an `undefined` value as in JavaScript, but this is one of those moments.)

Reply:
What I think would make sense: https://github.com/sindresorhus/query-string#falsy-values

```js
queryString.stringify({foo: false});
//=> 'foo=false'

queryString.stringify({foo: null});
//=> 'foo'

queryString.stringify({foo: undefined});
//=> ''
```

Reply:
Hmm, I wonder if the `httpx.UNSET` falls into the YAGNI category?

BTW Requests didn't allow setting session-level params from the constructor either (it should have though!) and only allows editing as a property. We can document that the params property has different behavior when it comes to unsetting a predefined value?

Reply:
> We can document that the params property has different behavior when it comes to unsetting a predefined value?

Our behavior being that we don't support unsetting? :)

Reply:
Hah, my wording was bad there but yes. Document that None adds a no-value entry to the query string instead of unsetting.

Reply:
We can leave this for the future. Maybe we should create an issue for it.

Reply:
Yes, I think that can be discussed and tackle separately. Feel fee to open the issue, or I’ll do it as part of post merge cleanup. :)